CoreScriptsRoot/Modules/Chat.lua

```diff
@@ -270,22 +270,30 @@ do
 		return nil -- Found no player
 	end
 
-	local adminCache = {}
-	function Util.IsPlayerAdminAsync(player)
-		local userId = player and player.userId
-		if userId then
-			if adminCache[userId] == nil then
-				local isAdmin = false
-				-- Many things can error is the IsInGroup check
-				pcall(function()
-					isAdmin = player:IsInGroup(1200769)
-				end)
-				adminCache[userId] = isAdmin
+	local function MakeIsInGroup(groupId, requiredRank)
+		assert(type(requiredRank) == "nil" or type(requiredRank) == "number", "requiredRank must be a number or nil")
+		
+		local inGroupCache = {}
+		return function(player)
+			if player and player.userId then
+				local userId = player.userId
+
+				if inGroupCache[userId] == nil then
+					local inGroup = false
+					pcall(function() -- Many things can error is the IsInGroup check
+						inGroup = (not requiredRank) and player:IsInGroup(groupId) or (player:GetRankInGroup(groupId) > requiredRank)
```

Comment:
As much as your psuedo-ternary operator coding is appreciated, I think it would be more straightforward just to do:

if requiredRank then
  inGroup = player:GetRankInGroup(groupId) > requiredRank
else
  inGroup = player:IsInGroup(groupId)
end